pkg/controller/v1alpha1/utils/utils.go

```diff
@@ -30,3 +41,106 @@ func CheckNodeAffinity(pvSpec *corev1.PersistentVolumeSpec, node corev1.Node) (b
+		"max-scale", annotations[autoscaling.MaxScaleAnnotationKey],
+	)
+
+	annotations[constants.InitialScaleAnnotationKey] = annotations[autoscaling.MinScaleAnnotationKey]
+
+	log.Info("kserve will always set the initial scale value for knative revisions equal to the min replicas requested",
+		"initial-scale", annotations[constants.InitialScaleAnnotationKey],
+	)
+
+	// Retrieve the allow-zero-initial-scale value from the knative autoscaler configuration.
+	allowZeroInitialScale, err := CheckZeroInitialScaleAllowed(ctx, client)
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve the knative autoscaler configuration")
+	}
+
+	// If knative is configured to not allow zero initial scale when 0 min replicas are requested,
+	// set the initial scale to 1 for the created knative revision. This represents the closest supported value
+	// to the requested min replicas. After initialization, the knative revision will be scaled to 0.
+	if !allowZeroInitialScale && revisionMinScale == 0 {
+		log.Info("The current knative autoscaler global configuration does not allow zero initial scale. The knative revision will be initialized with 1 replica then scaled down to 0",
+			"allow-zero-initial-scale", allowZeroInitialScale,
+			"initial-scale", 1)
+		annotations[constants.InitialScaleAnnotationKey] = "1"
+	}
+
+	return nil
+}
+
+// CheckZeroInitialScaleAllowed reads the global knative autoscaler configuration defined in the knativeserving
+// custom resource and returns a boolean value based on if knative is configured to allow zero initial scale.
+func CheckZeroInitialScaleAllowed(ctx context.Context, client client.Client) (bool, error) {
+	// Set allow-zero-initial-scale to the default values to start.
+	// If autoscaling values are not set in the configuration, then the defaults are used.
+	allowZeroInitialScale := "false"
+
+	// List all knativeserving custom resources.
+	// This allows for the handling of scenarios where the custom resource is not created
+	// in the default knative-serving namespace or without the default knative-serving name.
+	kservingList := &operatorv1beta1.KnativeServingList{}
+	err := client.List(ctx, kservingList)
+	if err != nil {
+		return false, errors.Wrapf(
+			err,
+			"fails to retrieve the knativeserving custom resource.",
+		)
+	}
+
+	if len(kservingList.Items) == 0 {
+		return false, errors.New("no knativeserving resources found in cluster.")
+	}
+
+	// Always use the first knativeserving resource returned.
+	// We are operating under the assumption that there should be a single knativeserving custom resource created on the cluster.
+	kserving := kservingList.Items[0]
+	if kserving.Spec.Config != nil {
+		if kservingAutoscalerConfig, ok := kserving.Spec.Config[constants.AutoscalerKey]; ok {
+			if configuredAllowZeroInitialScale, ok := kservingAutoscalerConfig[constants.AutoscalerAllowZeroScaleKey]; ok {
+				allowZeroInitialScale = configuredAllowZeroInitialScale
+			}
+		}
+	}
```

Comment:
Another downside of using KnativeServing CRD for this check is that users can set configurations for the autoscaler with both (equivalent with a priority for one)

```yaml
spec:
  config:
    autoscaler: ...
```
and
```yaml
spec:
  config:
    config-autoscaler: ...
```